File: GeneratedEntityReaderClasses/Components/ArcComponent.cs
namespace GeneratedNoitaClasses;
public class ArcComponent : NoitaComponentBase,  iNoitaType<ArcComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaUInt type { get; set; } = new();
	public NoitaInt material { get; set; } = new();
	public NoitaInt lifetime { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		type.Read(s);
		material.Read(s);
		lifetime.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		type.Write(s);
		material.Write(s);
		lifetime.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/ArcComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaUInt
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class ArcComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    type:NoitaUInt = field(default_factory=NoitaUInt)
    material:NoitaInt = field(default_factory=NoitaInt)
    lifetime:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.type.Read(s)
        self.material.Read(s)
        self.lifetime.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.type.Write(s)
        self.material.Write(s)
        self.lifetime.Write(s)

